deployments/ek-dashboards/es-sts.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: es-cluster
+  namespace: kube-system
+spec:
+  serviceName: elasticsearch
+  replicas: 3
+  selector:
+    matchLabels:
+      app: elasticsearch
+  template:
+    metadata:
+      labels:
+        app: elasticsearch
+    spec:
+      containers:
+      - name: elasticsearch
+        image: docker.elastic.co/elasticsearch/elasticsearch:7.2.0
+        resources:
+            limits:
+              cpu: 1000m
+            requests:
+              cpu: 100m
+        ports:
+        - containerPort: 9200
+          name: rest
+          protocol: TCP
+        - containerPort: 9300
+          name: inter-node
+          protocol: TCP
+        volumeMounts:
+        - name: data
+          mountPath: /usr/share/elasticsearch/data
+        env:
+          - name: cluster.name
+            value: k8s-logs
+          - name: node.name
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: discovery.seed_hosts
+            value: "es-cluster-0.elasticsearch,es-cluster-1.elasticsearch,es-cluster-2.elasticsearch"
+          - name: cluster.initial_master_nodes
+            value: "es-cluster-0,es-cluster-1,es-cluster-2"
+          - name: ES_JAVA_OPTS
+            value: "-Xms512m -Xmx512m"
+      initContainers:
+      - name: fix-permissions
+        image: busybox
+        command: ["sh", "-c", "chown -R 1000:1000 /usr/share/elasticsearch/data"]
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: data
+          mountPath: /usr/share/elasticsearch/data
+      - name: increase-vm-max-map
+        image: busybox
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+        securityContext:
+          privileged: true
+      - name: increase-fd-ulimit
+        image: busybox
+        command: ["sh", "-c", "ulimit -n 65536"]
+        securityContext:
+          privileged: true
+  volumeClaimTemplates:
+  - metadata:
+      name: data
+      labels:
+        app: elasticsearch
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      # storageClassName: ""
+      resources:
+        requests:
+          storage: 10Gi
```

Comment:
Do we need 10Gi? I am not sure what kind of memory do we require here.

Reply:
yes 10Gi loojs like a overkill. since there are 3 replicas , i think we can keep it at 5Gi each . Almost 1Gi space is for elasticsearch itself. 
disk requirement  depends on number of alerts to handle , on rough calulation at rate of 1000 alerts per hour and each alert consuming around 40KB, we can get decent time for almost a year on single continious use.

```
health status index     uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   .kibana_1 UhiacNwdSyCdwrZ_WZOdRg   1   1         26            0     70.3kb         35.1kb
green  open   alert     nai7xLdcRaKe-xStBA44qQ   1   1          4            0     77.4kb         38.7kb
```
here is a example , 4 alerts taking 80kb space finally

Reply:
```
health status index     uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   .kibana_1 UhiacNwdSyCdwrZ_WZOdRg   1   1         26            0     70.3kb         35.1kb
green  open   alert     nai7xLdcRaKe-xStBA44qQ   1   1          4            0     77.4kb         38.7kb
```

i have some real data here , take a count of 1000 alerts per hour and assume a storage capacity of total 12 GB for Elasticsearch , it can sustain approximately 27 days of storage.

Reply:
Ideally we expect the alerts to be less (otherwise something is really bad), I would say 1000 alerts/hr should be fine. 
@nyrahul wdyt?